ui/admin-portal/src/app/components/candidates/view/tasks/view-candidate-tasks.component.ts

```diff
@@ -102,6 +106,43 @@ export class ViewCandidateTasksComponent implements OnInit, OnChanges {
 
   }
 
+  assignDuolingoCouponTask() {
+    this.duolingoCouponService.assignCouponToCandidate(this.candidate.id).subscribe(
+      () => {
+        this.taskService.listTasks().subscribe(
+          tasks => {
+            const duolingoTaskId = tasks.find(task => task.name === 'duolingoTest')?.id;
```

Comment:
I think it will all work better and be simpler if the task assignment is automated on the server end whenever a candidate is assigned a coupon.

Then all this code goes - moved to the DuolingoCoupon server code. Maybe in the DuolingoCouponAdminApi or in the DuolingoCouponService.

Then we just have a Assign Duolingo Coupon button - and the server code can handle the way the candidate is notified - task, email - or both.

By the way where does the email notification get sent in the current code?